forge-kernel/src/tools/forge_verify.cpp

```
@@ -0,0 +1,744 @@
+bool probeDiscriminates(const forge::PointInSolid& probe, const forge::Mesh& mesh,
+                        const double lo[3], const double hi[3], double volume,
+                        std::string& why) {
+    double span = 0.0;
+    for (int k = 0; k < 3; ++k) span = std::max(span, hi[k] - lo[k]);
+    if (!(span > 0.0)) { why = "bounding box has no extent"; return false; }
+    const double eps = 1e-4 * span;
+
+    const auto& P = mesh.positions;
+    const auto& I = mesh.indices;
+    const std::size_t nTri = I.size() / 3;
+    if (nTri > 0) {
```

Clang (14.0.6)

[note] +1, including nesting penalty of 0, nesting level increased to 1

(clang)